### ISSUE TITLE
Calling reject() with no reason results in a successful activation result

### ISSUE THREAD

**Issue by l2fprod** [openwhisk/openwhisk#1562]
Consider this action:

```
function main(args) {
  return new Promise((resolve, reject) => {
    reject();
  });
}
```

When invoked, it results in a successful activation:

```
    "response": {
        "status": "success",
        "statusCode": 0,
        "success": true,
        "result": {}
    },
```

Although I would have expected it to fail as I called `reject`.

If I change `reject()` to `reject("with a reason")`, now it fails as expected:

```
    "response": {
        "status": "application error",
        "statusCode": 0,
        "success": false,
        "result": {
            "error": "with a reason"
        }
    },
```

It would be great if `reject()` would actually result in a failed execution.

**Comment by rabbah** [comment]
reject should return an {error: truthy} in this case.

**Comment by csantanapr** [comment]
I agree it's odd, we should look into it 
to see if we can support resolve() and reject() with no argument the proxy has the ability to detect this and handle it appropriately

**Comment by psuter** [comment]
Proposed fix in #1611: non-truthy errors are turned into `{}` (this includes `reject()`, which essentially maps to `return { error: undefined }`).

This is consistent with the current behavior of `return;` turning into `return {};`.
